src/cpp/session/modules/SessionHistory.cpp

```diff
@@ -361,12 +361,20 @@ void onHistoryAdd(const std::string& command)
    module_context::enqueClientEvent(event);
 }
 
-SEXP rs_timestamp(SEXP dateSEXP)
+SEXP rs_timestamp(SEXP stampSEXP, SEXP prefixSEXP, SEXP suffixSEXP, SEXP quietSEXP)
 {
-   boost::format fmt("##------ %1% ------##");
-   std::string ts = boost::str(fmt % r::sexp::safeAsString(dateSEXP));
+   std::string prefix = r::sexp::safeAsString(prefixSEXP);
```

Comment:
There is a risk here that the prefix or suffix will contain characters that will be interpreted by `boost::fmt` and cause the formatter to fail. We could escape them but I think we should just use regular string append/prepend here.